From: SeongJae Park <sj@kernel.org>
Date: Fri, 4 Jul 2025 12:20:50 -0700
Subject: [PATCH] mm/damon/core: destroy targets when kdamond_fn() finish

There is no reason to keep targets after kdamond_fn() is complete.
Destroy.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 3ad8c29946ad..30e09a9d1a56 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2816,6 +2816,7 @@ static int kdamond_fn(void *data)
 		running_exclusive_ctxs = false;
 	mutex_unlock(&damon_lock);
 
+	damon_destroy_targets(ctx);
 	return 0;
 }
 
-- 
2.39.5

